ocular/src/chain/client/query.rs

```diff
@@ -1,9 +1,145 @@
-use crate::error::{ChainClientError, RpcError};
+use crate::{
+    cosmos_modules::{base::Coin, *},
+    error::{ChainClientError, GrpcError, RpcError},
+};
 use tendermint_rpc::Client;
+use tonic::transport::Channel;
 
 use super::ChainClient;
 
+pub type AuthQueryClient = auth::query_client::QueryClient<Channel>;
+pub type BankQueryClient = bank::query_client::QueryClient<Channel>;
+pub type DistributionQueryClient = distribution::query_client::QueryClient<Channel>;
+pub type EvidenceQueryClient = evidence::query_client::QueryClient<Channel>;
+pub type GovQueryClient = gov::query_client::QueryClient<Channel>;
+pub type MintQueryClient = mint::query_client::QueryClient<Channel>;
+pub type ParamsQueryClient = params::query_client::QueryClient<Channel>;
+pub type SlashingQueryClient = slashing::query_client::QueryClient<Channel>;
+pub type StakingQueryClient = staking::query_client::QueryClient<Channel>;
+
 impl ChainClient {
+    fn check_for_grpc_address(&self) -> Result<(), GrpcError> {
+        if self.config.grpc_address.is_empty() {
+            return Err(GrpcError::MissingEndpoint(format!(
+                "no grpc address available for chain {}",
+                self.config.chain_id
+            )));
+        }
+
+        Ok(())
+    }
+
+    // Auth queries
+    pub async fn get_auth_query_client(&self) -> Result<AuthQueryClient, ChainClientError> {
+        self.check_for_grpc_address()?;
+
+        AuthQueryClient::connect(self.config.grpc_address.clone())
+            .await
+            .map_err(|e| GrpcError::Connection(e).into())
+    }
+
+    // Bank queries
+    pub async fn get_bank_query_client(&self) -> Result<BankQueryClient, ChainClientError> {
+        self.check_for_grpc_address()?;
+
+        BankQueryClient::connect(self.config.grpc_address.clone())
+            .await
+            .map_err(|e| GrpcError::Connection(e).into())
+    }
```

Comment:
A couple things we could do here to make this more flexible (or we can defer this for later if we don't want to do it as part of this PR):

* Loop through a list of possible endpoints if some are having issues/failing
* Add in a backoff retry mechanism for failed connections

Reply:
That sounds good to me! I'll add another issue to the backlog for that

Reply:
Created https://github.com/PeggyJV/ocular/issues/23